src/stacks-hierarchy/Root_TicketAssistantStack/TicketAssistant_DurationScreen/TicketAssistant_DurationScreen.tsx

```diff
@@ -0,0 +1,358 @@
+import {Button} from '@atb/components/button';
+import React, {useEffect, useState} from 'react';
+import {
+  sliderColorMax,
+  sliderColorMin,
+} from '@atb/stacks-hierarchy/Root_TicketAssistantStack/TicketAssistant_FrequencyScreen';
+import {TicketAssistantScreenProps} from '@atb/stacks-hierarchy/Root_TicketAssistantStack/navigation-types';
+import {useTicketAssistantState} from '@atb/stacks-hierarchy/Root_TicketAssistantStack/TicketAssistantContext';
+import RNDateTimePicker from '@react-native-community/datetimepicker';
+import {useLocaleContext} from '@atb/LocaleProvider';
+import {ThemeIcon} from '@atb/components/theme-icon';
+import SvgDate from '@atb/assets/svg/mono-icons/time/Date';
+import {format, parseISO} from 'date-fns';
+import {dateToDateString} from '@atb/components/sections/items/date-input/utils';
+import {useAccessibilityContext} from '@atb/AccessibilityContext';
+import {SectionSeparator} from '@atb/components/sections';
+import {
+  dateDiffInDays,
+  getDateFromSlider,
+  getDaysWeeksMonths,
+  getSliderIndex,
+} from './utils';
+type DurationProps =
+  TicketAssistantScreenProps<'TicketAssistant_DurationScreen'>;
+
+const currentDate = new Date();
+const durations = [1, 7, 14, 21, 30, 60, 90, 120, 150, 180];
+
+export const TicketAssistant_DurationScreen = ({navigation}: DurationProps) => {
+  const {data, updateData} = useTicketAssistantState();
+  const [date, setDate] = useState(dateToDateString(currentDate));
+  const [showDatePicker, setShowDatePicker] = useState(false);
+  const [usedSlider, setUsedSlider] = useState(false);
+  const styles = useThemeStyles();
+  const {t} = useTranslation();
+  const a11yContext = useAccessibilityContext();
+  const [sliderValue, setSliderValue] = useState(data.duration);
+
+  useEffect(() => {
+    const unsubscribe = navigation.addListener('blur', () => {
+      if (usedSlider) {
+        updateData({...data, duration: sliderValue});
+      }
+    });
+    return () => {
+      unsubscribe();
+    };
+  }, [navigation, data, sliderValue, updateData]);
+
+  const majorVersionIOS = parseInt(String(Platform.Version), 10);
+  const style =
+    majorVersionIOS < 13 ? {width: undefined, flex: 1} : {width: 130};
+
+  const locale = useLocaleContext();
+
+  function updateDuration(value: number, fromPicker?: boolean) {
+    const newData = {...data, duration: fromPicker ? value : durations[value]};
+    updateData(newData);
+    setSliderValue(newData.duration);
+  }
+
+  const duration = usedSlider ? sliderValue : data.duration;
+  const daysWeeksMonths = getDaysWeeksMonths(duration);
+  const resultString = t(
+    duration < 7
+      ? TicketAssistantTexts.duration.resultDays({value: daysWeeksMonths})
+      : duration < 30
+      ? TicketAssistantTexts.duration.resultWeeks({value: daysWeeksMonths})
+      : duration < 180
+      ? TicketAssistantTexts.duration.resultMonths({value: daysWeeksMonths})
+      : TicketAssistantTexts.duration.resultMoreThan180Days,
+  );
```

Comment:
You have mainly the same logic inside `getDaysWeeksMonths` and the logic for the `resultString`, where both checks if duration is greater than 7, 30, etc. Can it be consolidated?